### ISSUE TITLE
Inline widget bug in Firefox

### ISSUE THREAD

**Issue by ddmm-white** [ckeditor/ckeditor5#6532]
Hi. There is a weird issue with plugins that use inline widget in Firefox. If I press Space bar right after or before an inline widget, the widget disappears. Also it's not possible to place the caret after an inline widget using mouse. You can only do that with Right arrow key. I've recorded some screencasts for you to see what I mean:
Firefox:
![recording2](https://user-images.githubusercontent.com/9927064/78211711-6d9b8500-74b6-11ea-9506-db0347a90a4b.gif)

Chrome:
![recording3](https://user-images.githubusercontent.com/9927064/78211780-a63b5e80-74b6-11ea-8503-fb3a0d6880ce.gif)

As you can see, in Firefox it has completely different behavior. In Chrome it works fine.
JSFiddle: https://jsfiddle.net/lastMorning/1yq09k5s
Sources of custom build with this plugin: https://github.com/ddmm-white/ckeditor5-build-classic

**Comment by Mgsy** [comment]
@jodator, can you take a look at it?

**Comment by ddmm-white** [comment]
Hi again. Can someone help with this please? I couldn't find a fix to "spacebar after formula" glitch on my own.

**Comment by jodator** [comment]
Hi @ddmm-white  it looks like your code is missing [position mapping fixing](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/tutorials/implementing-an-inline-widget.html#fixing-position-mapping).

**Comment by ddmm-white** [comment]
I updated my build and plugin, added position mapping fix and changed container element to _span_. But now if I enter any text after the widget, the widget turns into plain text. Also I noticed that if I create the same element (with only text inside, without UIElement) for both editingDowncast and dataDowncast converters then it starts working. But i can't do that for my custom plugin because I need UIElement for the editingDowncast converter.

**Comment by jodator** [comment]
> I need UIElement for the editingDowncast converter.

Could you explain more? To be honest the inline widget sample is currently the only use-case that we've tested. The [`UIElement`](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_view_uielement-UIElement.html) is meant to be used as elements that are not mapped to the model content of the editor. So everything that helps in editing but is not part of the data - e.g. Widget's selector handle.

In other words, there must be some `ViewElement` in the view so the model \<-> view mappings work properly for calculating positions, etc.

So maybe you wish to accomplish something that `UIElement` do similarly but without expected behavior of inline widget it is hard to tell.

**Comment by FilipTokarski** [comment]
I'm closing this due to lack of activity.

**Comment by petrovich2k** [comment]
It seems that the issue is still valid in CkEditor v34. I'm able to reproduce it in Firefox 100.0.2. I added 3 inline images, but I was unable to place the caret between them by clicking. 
![mouse](https://user-images.githubusercontent.com/72270591/170389799-1cecf528-18f1-438f-b1ba-7c815f8ec538.gif)

However, I could place the caret between images when I navigated with arrow key on my keyboard.
![keyboard](https://user-images.githubusercontent.com/72270591/170389904-9a837298-d24e-4210-b05d-6577c4bf80e5.gif)

There is a workaround. If I add spaces between the images from keyboard, I can place the caret between the images with mouse.
![space](https://user-images.githubusercontent.com/72270591/170390040-18ed22e4-83c4-4138-bec0-5be13b61f1e6.gif)
